### PR TITLE
ROX-23028: Add page component layout for node cves page

### PR DESCRIPTION
**Pull request by dvail**
## Description

Depends on refactoring in https://github.com/stackrox/stackrox/pull/10355

- Adds the main layout of the Node CVE list page.
- Basic integration of entity tabs, filters, and pagination with URL parameters

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Visit Node CVEs from the sidebar and see the rendered page layout. The correct entityTab parameter should be visible in the URL:
image

Change to the Node entity and ensure the correct entityTab parameter is visible in the URL:
image

Apply various filters and change the "perPage" setting and ensure the URL tracks these values:
image

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by dvail** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10356?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10356** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10356?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#10355** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10355?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10315** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10315?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10304** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10304?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dvail and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: ui/apps/platform/src/Containers/Vulnerabilities/components/FilterDropdowns.css [hunk 1]
```diff
@@ -1,4 +1,4 @@
 /* Fixes the issue where badge counts in select components wrap to a new line */
-.workload-table-toolbar .pf-c-select__toggle-wrapper {
+.vm-filter-toolbar-dropdown .pf-c-select__toggle-wrapper {
   flex-wrap: nowrap;
 }
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/components/FilterDropdowns.css:4]: This CSS was moved from the WorkloadCVE specific toolbar to the general component directory and applied directly to the two dropdowns that required it. This prevents the need for classes on the `<Toolbar>` components and avoids cross-importing on Workload/Node CVE pages.

File: ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/Overview/NodeCvesOverviewPage.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import EntityTypeToggleGroup from '../../components/EntityTypeToggleGroup';
+import NodeCveFilterToolbar from '../components/NodeCveFilterToolbar';
+import { NODE_CVE_SEARCH_OPTION } from '../../searchOptions';
+import { parseWorkloadQuerySearchFilter } from '../../utils/searchUtils';
+import { nodeEntityTabValues } from '../../types';
+
+import CVEsTableContainer from './CVEsTableContainer';
+import NodesTableContainer from './NodesTableContainer';
+
+const searchOptions = [NODE_CVE_SEARCH_OPTION];
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/Overview/NodeCvesOverviewPage.tsx:27]: More options will be added here in a follow up.

File: ui/apps/platform/src/Containers/Vulnerabilities/types.ts [hunk 3]
```diff
@@ -66,7 +66,15 @@ export function isValidWorkloadEntityTab(value: unknown): value is WorkloadEntit
     return workloadEntityTabValues.some((tab) => tab === value);
 }
 
-export type EntityTab = WorkloadEntityTab;
+export const nodeEntityTabValues = ['CVE', 'Node'] as const;
+
+export type NodeEntityTab = (typeof nodeEntityTabValues)[number];
+
+export function isValidNodeEntityTab(value: unknown): value is NodeEntityTab {
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/Vulnerabilities/types.ts:73]: I noticed this function and `isValidWorkloadEntityTab` aren't being used. Will they be used later or did it turn out they weren't needed?

**Review by sachaudh — APPROVED** [review]

File: ui/apps/platform/src/Containers/Vulnerabilities/types.ts (hunk 3 above)
Comment (dvail) [reply ui/apps/platform/src/Containers/Vulnerabilities/types.ts:73]: Hmm I guess they aren't needed. I think `isValidWorkloadEntityTab` may have been used at some point, but no longer is.

I can clean these out in a follow up.

**Pull request merged**